packages/phone-number-privacy/signer/test/integration/pnp.test.ts

```diff
@@ -225,33 +264,544 @@ describe('pnp', () => {
+      quotaCalculationTestCases.forEach(({ expectedTotalQuota, cusdOdisPaymentInWei }) => {
+        it(`Should get totalQuota=${expectedTotalQuota} 
+           for cUSD (wei) payment=${cusdOdisPaymentInWei.toString()}`, async () => {
+          mockOdisPaymentsTotalPaidCUSD.mockReturnValue(cusdOdisPaymentInWei)
+
+          const req = getPnpSignRequest(
+            ACCOUNT_ADDRESS1,
+            BLINDED_PHONE_NUMBER,
+            AuthenticationMethod.WALLET_KEY,
+            IDENTIFIER
+          )
+          const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+          const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
+
+          const shouldSucceed = expectedTotalQuota > 0
+
+          if (shouldSucceed) {
+            expect(res.status).toBe(200)
+            expect(res.body).toMatchObject<SignMessageResponseSuccess>({
+              success: true,
+              version: expectedVersion,
+              signature: expectedSignature,
+              performedQueryCount: 1, // incremented for signature request
+              totalQuota: expectedTotalQuota,
+              blockNumber: testBlockNumber,
+              warnings: [],
+            })
+          } else {
+            expect(res.status).toBe(403)
+            expect(res.body).toMatchObject<SignMessageResponseFailure>({
+              success: false,
+              version: expectedVersion,
+              performedQueryCount: 0,
+              totalQuota: expectedTotalQuota,
+              blockNumber: testBlockNumber,
+              error: WarningMessage.EXCEEDED_QUOTA,
+            })
+          }
+        })
+      })
+    })
+
+    describe('endpoint functionality', () => {
+      // Use values already tested in quota logic tests, [onChainBalance, expectedQuota]
+      const performedQueryCount = 2
+
+      beforeEach(async () => {
+        mockOdisPaymentsTotalPaidCUSD.mockReturnValue(onChainBalance)
+        await db.transaction(async (trx) => {
+          for (let i = 0; i < performedQueryCount; i++) {
+            await incrementQueryCount(db, ACCOUNT_ADDRESS1, rootLogger(config.serviceName), trx)
+          }
+        })
+      })
+
+      it('Should respond with 200 on valid request', async () => {
+        const req = getPnpSignRequest(
+          ACCOUNT_ADDRESS1,
+          BLINDED_PHONE_NUMBER,
+          AuthenticationMethod.WALLET_KEY,
+          IDENTIFIER
+        )
+        const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+        const res = await sendRequest(req, authorization, SignerEndpoint.PNP_SIGN)
+        expect(res.status).toBe(200)
+        expect(res.body).toMatchObject<SignMessageResponseSuccess>({
+          success: true,
+          version: res.body.version,
+          signature: expectedSignature,
+          performedQueryCount: performedQueryCount + 1,
+          totalQuota: expectedQuota,
+          blockNumber: testBlockNumber,
+          warnings: [],
+        })
```

Comment:
same as in legacypnp tests, maybe worth adding a check for the expected KEY_VERSION_HEADER value when none was provided in the req?